src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let value = 0;
+
+  const adder = (a) => {
+    const count = value;
+
+    if (a !== undefined) {
+      value += a;
+
+      return adder;
+    } else {
+      value = 0;
+
+      return count;
```

Comment:
The current logic resets `value` to 0 when the function is called without arguments, which means the cumulative sum is lost after the first retrieval. If the intention is to keep the cumulative sum, consider removing the line `value = 0;`.